### ISSUE TITLE
[False Positive] Integer overflow warning due to precision loss in closures

### ISSUE THREAD

**Issue by lynae99** [lizhuohua/rust-mir-checker#24]
`MirChecker` reports a false positive integer overflow warning. The tool fails to track the return value of a function when the calculation is wrapped in a closure, treating the value as "Unknown" (Top). This causes safe arithmetic operations downstream to be flagged as potentially overflowing.
**Reproduction**
```
fn side_effect_return(a: i32) -> i32 {
    (|| {
        2 * a
    })()
}

fn side_effect_arg(a: &mut i32) {
    *a = *a + 1;
}

#[allow(unused_variables)]
fn main() {
    let mut r = 5;
    r = side_effect_return(3);
    
    side_effect_arg(&mut r);
}
```
**Actual Behavior**
The tool reports a possible overflow error for a safe operation:
```
warning: [MirChecker] Possible error: "attempt to compute `\"move _2\" + \"const 1_i32\"`, which would overflow"
 --> src/main.rs:8:10
  |
8 |     *a = *a + 1;
  |          ^^^^^^
```
**Expected Behavior**
No warning should be reported. The actual value of `r` is `6` (from `3 * 2`), and `6 + 1` is safe from overflow.
